src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking.cs

```diff
@@ -35232,25 +35262,30 @@ public class B<T> where T : A
 {
     static void Main()
     {
-        new B<A?>();
-        new B<A>();
+        new B1<A1?>();
```

Comment:
> new B1<A1?>(); [](start = 7, length = 15)

This keeps throwing me off (I expect there would be an error, since `T` is constrained to `A1` (which looks non-nullable, so my intuition is "this is a non-nullable constraint"). 
I think there is a usability problem which should be re-discussed.

Reply:
Thinking more about this, maybe this is just a problem with me. When I start thinking of the relationship `A1?` is-a `A1`, then the confusion lifts...

---
In reply to: [199555426](https://github.com/dotnet/roslyn/pull/28239#discussion_r199555426) [](ancestors = 199555426)